index.bs

```diff
@@ -252,6 +272,125 @@ pause).
+Picture-in-Picture mode to only one window.  Whether only one window is allowed
+in Picture-in-Picture mode will be left to the implementation and the platform.
+However, because of the one Picture-in-Picture window limitation, the
+specification assumes that a given {{Document}} can only have one
+Picture-in-Picture window.
+
+What happens when there is a Picture-in-Picture request while a window is
+already in Picture-in-Picture will be left as an implementation detail: the
+current Picture-in-Picture window could be closed, the Picture-in-Picture
+request could be rejected or even two Picture-in-Picture windows could be
+created. Regardless, the User Agent will have to fire the appropriate events
+in order to notify the website of the Picture-in-Picture status changes.
+
+# Arbitrary Content Specific Concepts # {#arbitrary-concepts}
+
+The Picture-in-Picture window MAY support <dfn>arbitrary content</dfn> which will
+display a {{Document}} in the Picture-in-Picture window. The <a>arbitrary content</a>
+MAY be <dfn>interactive arbitrary content</dfn> which will allow user interaction.
+
+The <a>arbitrary content</a> MAY also be <dfn>non-interactive arbitrary content</dfn>
+which will not allow user interaction. However, the user agent MAY show
+Picture-in-Picture window controls to support some interaction.
+
+Whether <a>interactive arbitrary content</a> is supported in Picture-in-Picture mode
+is determined by the platform and will be left to the platform.
+
+Whether the <a>arbitrary content</a> is interactive or not will be specified by the
+website. <!-- TODO(beccahughes): Reference how this done -->
+
+## Request Element Picture-in-Picture ## {#request-element-pip}
+
+When the <dfn>request element Picture-in-Picture algorithm</dfn> with |element|
+and |interactive| is invoked, the user agent MUST run the following steps:
+
+1. Run the <a>check Picture-in-Picture is allowed algorithm</a> with |element|.
+    Abort these steps if an error was thrown.
+2. If |interactive| is `true` and the platform does not support
+    <a>interactive arbitrary content</a>, throw a {{NotSupportedError}} and abort
+    these steps.
+3. If the algorithm is not <a>triggered by user activation</a>, throw a
+    {{NotAllowedError}} and abort these steps.
+4. If |element| is {{pictureInPictureElement}}, abort these steps.
+6. Let |pictureInPictureBrowsingContext| be a new <a>browsing context</a> with the
+    <a>opener browsing context</a> set to the current document's browsing context.
+7. Let |document| be a new {{Document}} with its browsing context set to
+    |pictureInPictureBrowsingContext| and the URL set to <a>about:blank</a>.
+8. Let |clone children flag| be `true`.
+9. Run <a>node clone</a> with |element|, |document| and |clone children flag|.
+10. Let <a>Picture-in-Picture window</a> be a new instance of
+    {{PictureInPictureWindow}} associated with |element| and with a reference to
+    |document|.
```

Comment:
Should this be part of another PR as there is no `document` yet. WDYT?

Reply:
We define this in https://github.com/WICG/picture-in-picture/pull/136 so we could always wait to submit that PR before this one